onadata/apps/main/management/commands/update_enketo_urls.py

```diff
@@ -45,25 +48,45 @@ def handle(self, *args, **options):
-        for meta_data in resultset:
-            username = meta_data.content_object.user.username
-            id_string = meta_data.content_object.id_string
-            data_type = meta_data.data_type
-            data_value = meta_data.data_value
-            xform = meta_data.content_object
-            with open('/tmp/enketo_url', 'a') as f:
-
-                if data_type == 'enketo_url':
-                    form_url = get_form_url(
-                        request, username, protocol=protocol,
-                        xform_pk=xform.pk)
-                    _enketo_url = enketo_url(form_url, id_string)
-                    MetaData.enketo_url(xform, _enketo_url)
-                elif data_type == 'enketo_preview_url':
-                    _enketo_preview_url = get_enketo_preview_url(
-                        request, username, id_string, xform_pk=xform.pk)
-                    MetaData.enketo_preview_url(xform, _enketo_preview_url)
-                f.write('%s : %s \n' % (id_string, data_value))
+
+        if use_username:
+            for meta_data in resultset:
+                username = meta_data.content_object.user.username
+                id_string = meta_data.content_object.id_string
+                data_type = meta_data.data_type
+                data_value = meta_data.data_value
+                xform = meta_data.content_object
+                with open('/tmp/enketo_url', 'a') as f:
+
+                    if data_type == 'enketo_url':
+                        form_url = get_form_url(
+                            request, username, protocol=protocol,
+                            xform_pk=xform.pk)
+                        _enketo_url = enketo_url(form_url, id_string)
+                        MetaData.enketo_url(xform, _enketo_url)
+                    elif data_type == 'enketo_preview_url':
+                        _enketo_preview_url = get_enketo_preview_url(
+                            request, username, id_string, xform_pk=xform.pk)
+                        MetaData.enketo_preview_url(xform, _enketo_preview_url)
+                    f.write('%s : %s \n' % (id_string, data_value))
+        else:
+            for meta_data in resultset:
+                data_type = meta_data.data_type
+                data_value = meta_data.data_value
+                xform = meta_data.content_object
+
+                with open('/tmp/enketo_url', 'a') as f:
+                    if data_type == 'enketo_url':
+                        form_url = get_form_url(
+                            request, protocol=protocol,
+                            xform_uuid=xform.uuid)
+                        _enketo_url = enketo_url(form_url, id_string)
+                        MetaData.enketo_url(xform, _enketo_url)
+                    elif data_type == 'enketo_preview_url':
+                        _enketo_preview_url = get_enketo_preview_url(
+                            request, id_string, xform_uuid=xform.uuid)
```

Comment:
where is `id_string` defined?

Reply:
The `id_string` was defined at the beginning of the `for` loop.  I've changed how it's being defined in the latest commits. Here is the [updated definition of the variable](https://github.com/onaio/onadata/blob/da4a526eb15e9e06a8da91563e6c4bcccd833239/onadata/apps/main/management/commands/update_enketo_urls.py#L53-L58)